syntax = "proto3";

package helloworld;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloStream(HelloRequest) returns (stream HelloStreamReply);
  rpc SayHelloNested(HelloRequest) returns (HelloNestedReply);
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  repeated string message = 1;
}

message HelloStreamReply  {
  string message = 1;
}

message FooBar {
  string foo = 1;
  string bar = 2;
}

message HelloNestedReply {
  repeated FooBar message = 1;
}
